interface/forms/LBF/new.php

```diff
@@ -738,41 +1070,361 @@ function validate(f) {
+    echo "  <td class='bold' align='right'>" . xlt('Delete'  ) . "</td>\n";
+    echo " </tr>\n";
+    $fs->loadProductItems();
+    foreach ($fs->productitems as $lino => $li) {
+        echo " <tr>\n";
+        echo "  <td class='text'>" . text($li['code_text']) . "&nbsp;</td>\n";
+        echo "  <td class='text'>" .
+          $fs->genWarehouseSelect("form_fs_prod[$lino][warehouse]", '', $li['warehouse'], false, $li['hidden']['drug_id'], true) .
+          "  &nbsp;</td>\n";
+        echo "  <td class='text' align='right'>" .
+          "<input type='text' name='form_fs_prod[$lino][units]' size='3' value='" . $li['units'] . "' />" .
+          "&nbsp;</td>\n";
+        echo "  <td class='text' align='right'>" . oeFormatMoney($li['price']) . "&nbsp;</td>\n";
+        echo "  <td class='text' align='right'>\n" .
+          "   <input type='checkbox' name='form_fs_prod[$lino][del]' " .
+          "value='1'" . ($li['del'] ? " checked" : "") . " />\n";
+        foreach ($li['hidden'] as $hname => $hvalue) {
+            echo "   <input type='hidden' name='form_fs_prod[$lino][$hname]' value='" . attr($hvalue) . "' />\n";
+        }
+        echo "  </td>\n";
+        echo " </tr>\n";
+    }
+    echo "</table>\n";
+    echo "</center>\n";
+    echo "</div>\n";
+
+} // End Products Section
+
+if (isset($LBF_DIAGS_SECTION)) {
+
+    // Create the checkbox and div for the Diagnoses Section.
+    echo "<br /><span class='bold'><input type='checkbox' name='form_cb_fs_diags' value='1' " .
+      "onclick='return divclick(this, \"div_fs_diags\");'";
+    if ($display_style == 'block') echo " checked";
+    echo " /><b>" . xlt('Diagnoses') . "</b></span>\n";
+    echo "<div id='div_fs_diags' class='section' style='display:" . attr($display_style) . ";'>\n";
+    echo "<center>\n";
+    $display_style = 'none';
+
+    // If there are associated codes, generate a checkbox for each one.
+    if ($LBF_DIAGS_SECTION) {
+        echo "<table cellpadding='0' cellspacing='0' width='100%'>\n";
+        $cols = 3;
+        $tdpct = (int) (100 / $cols);
+        $count = 0;
+        $relcodes = explode(';', $LBF_DIAGS_SECTION);
+        foreach ($relcodes as $codestring) {
+            if ($codestring === '') continue;
+            $codes_esc = htmlspecialchars($codestring, ENT_QUOTES);
+            $cbval = $fs->genCodeSelectorValue($codestring);
+            if ($count % $cols == 0) {
+                if ($count) echo " </tr>\n";
+                echo " <tr>\n";
+            }
+            echo "  <td width='$tdpct%'>";
+            echo "<input type='checkbox' id='form_fs_diags[$codes_esc]' " .
+              "onclick='fs_diag_clicked(this)' value='" . attr($cbval) . "'";
+            if ($fs->code_is_in_fee_sheet) echo " checked";
+            list($codetype, $code) = explode(':', $codestring);
+            $crow = sqlQuery("SELECT code_text FROM codes WHERE " .
+              "code_type = ? AND code = ? AND active = 1 " .
+              "ORDER BY id LIMIT 1",
```

Comment:
What kind of codes is this for. If possibly includes external type codes, then should be using function in custom/code_types.inc.php
(in this case, the lookup_code_descriptions() function)

Reply:
Thanks (as above).